zetaclient/chains/bitcoin/observer/observer.go

```diff
@@ -341,20 +342,9 @@ func (ob *Observer) WatchGasPrice() {
 // PostGasPrice posts gas price to zetacore
 // TODO(revamp): move to gas price file
 func (ob *Observer) PostGasPrice() error {
-	// hardcode gas price here since this RPC is not available on regtest
-	if chains.IsBitcoinRegnet(ob.Chain().ChainId) {
-		blockNumber, err := ob.btcClient.GetBlockCount()
-		if err != nil {
-			return err
-		}
-
-		// #nosec G701 always in range
-		_, err = ob.ZetacoreClient().PostGasPrice(ob.Chain(), 1, "100", uint64(blockNumber))
-		if err != nil {
-			ob.logger.GasPrice.Err(err).Msg("PostGasPrice:")
-			return err
-		}
-		return nil
+	// special handle regnet and testnet gas rate
+	if ob.Chain().NetworkType != chains.NetworkType_mainnet {
+		return ob.specialHandleFeeRate()
```

Comment:
NIT: I think it might be easier to read if post gas price from just one point in the code, irrespective of testnet ,private or mainnet,  basically have this function return the rate instead of posting directly

Reply:
https://github.com/zeta-chain/node/pull/2452/commits/3dc39b63a44ac0eefa8b765b4120fc946d5f32b5